docs/architecture/adr-023-gas-used-and-gas-price-estimation.md

```diff
@@ -0,0 +1,113 @@
+# ADR 023: Gas used and gas price estimation
+
+## Changelog
+
+- 2025/01/17: Initial draft (@rach-id)
+
+## Status
+
+Proposed
+
+## Context
+
+As per [CIP-18: Standardised Gas and Pricing Estimation Interface](https://github.com/celestiaorg/CIPs/blob/main/cips/cip-18.md), celestia-app will provide a base implementation for the gas and fee estimation so it can be used when building transactions.
+
+## Decision
+
+We will proceed with the implementation in celestia-app.
+
+The fee estimation mechanism is TBD.
```

Comment:
Is this still TBD? This ADR describes two different fee estimation mechanisms so seems like we can remove the TBD.

```suggestion
The fee estimation mechanism is described below.
```